lib/puppet-languageserver/message_handler.rb

```diff
@@ -78,6 +78,23 @@ def request_puppet_compilenodegraph(_, json_rpc_message)
       end
     end
 
+    def request_puppetfile_getdependencies(_, json_rpc_message)
+      file_uri = json_rpc_message.params['uri']
+      return LSP::PuppetfileDependencyResponse.new('error' => 'Must be a puppetfile in order to find dependencies.') unless documents.document_type(file_uri) == :puppetfile
+
+      content = documents.document(file_uri)
+
+      result = []
+      begin
+        result = PuppetLanguageServer::Puppetfile::ValidationProvider.find_dependencies(content)
+      rescue StandardError => e
+        PuppetLanguageServer.log_message(:error, "(puppetfile/getdependencies) Error parsing puppetfile. #{e}")
+        return LSP::PuppetfileDependencyResponse.new('error' => 'An internal error occured while parsing the puppetfile. Please see the debug log files for more information.')
```

Comment:
This may not be best message for an invalid puppetfile. It is possible to rescue particular error classes.

Reply:
Will be addressed in https://github.com/puppetlabs/puppet-editor-services/issues/257